### PR TITLE
extend documentation

### PR DESCRIPTION
**Pull request by Borda**
Improving the documentation. In particular
- make left content table more granular
- top titles can be linked to left menu items

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

the actual version is https://torch-lightning.readthedocs.io/en/docs

### CODE REVIEW

**Comment by Borda** [comment]
@williamFalcon @jeffling @Ir1d @neggert ^^

File: docs/source/new-project.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+Quick Start
+===========
+To start a new project define two files, a LightningModule and a Trainer file.    
+To illustrate Lightning power and simplicity, here's an example of a typical research flow.
```
Comment (jeffling) [inline docs/source/new-project.rst]: `to illustrate Lightning's`

Comment (Borda) [reply docs/source/new-project.rst]: well there is not much new text (almost none)

**Comment by Borda** [comment]
> These changes are a net good, though I found a small grammar error.

Not good? so what other changes do you recommend?

> Reading through the docs, I think the
> formatting is broken.
> image

is it a specific error or is it on multiple places?

**Comment by jeffling** [comment]
I couldn't find any other examples of that formatting, so maybe it's just isolated

**Pull request merged**